service/instance.py

```diff
@@ -963,7 +958,6 @@ def launch_volume_instance(driver, identity, volume, size, name,
     """
     kwargs, userdata, network = _pre_launch_instance(
         driver, identity, size, name, **kwargs)
-    kwargs.update(prep_kwargs)
```

Comment:
badly cut and pasted code.  It seems that this was cut and pasted from lines 980-986 (see below).  prep_kwargs was set where kwargs is set.   Within this context there is no need update the kwargs  as they are updated in the _pre_launch_instance already.